machine/v1beta1/types_awsprovider.go

```diff
@@ -84,6 +84,13 @@ type AWSMachineProviderConfig struct {
 	// When omitted, no placement group is used when creating the EC2 instance.
 	// +optional
 	PlacementGroupName string `json:"placementGroupName,omitempty"`
+	// placementGroupPartition is the partition number within the placement group in which to launch the instance.
+	// This must be an integer value between 1 and 7. It is only valid if the placement group, referred in
+	// `PlacementGroupName` was created with strategy set to partition.
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=7
```

Comment:
@JoelSpeed I was doing some dev testing, and it turns out that these kubebuilder validations are not working as expected while creating a `MachineSet` object. Do we need them to be added inside the webhook?

Reply:
Yes, kubebuilder validations won't ever work on this particular API, please add them to the webhook validations

Reply:
Gotcha! Then these kubebuilder validations can be removed from here I think. Created https://github.com/openshift/api/pull/1942 to drop them

Reply:
https://github.com/openshift/machine-api-operator/pull/1265 to add them into webhook validations